sdk-cpp/tests/download_properties_tests.cpp

```diff
@@ -236,13 +236,35 @@ TEST_F(DownloadPropertyTests, BasicRangesTest)
     ASSERT_EQ(downloadedBytes, 1100);
 }
 
+TEST_F(DownloadPropertyTests, BasicEnumDownloadsTest)
+{
+    auto download1 = msdot::download::make(g_smallFileUrl, g_tmpFileName);
+
+    std::string downloadId1;
+    download1->get_property(msdo::download_property::id, downloadId1);
+
+    std::vector<std::unique_ptr<msdo::download>> downloads;
+    ASSERT_EQ(msdo::download::get_downloads(msdo::download_property::id, downloadId1, downloads).value(), 0);
+    ASSERT_EQ(downloads.size(), 1);
+
+    auto download2 = std::make_unique<msdot::download>(std::move(downloads[0]));
+    std::string downloadId2;
+    download2->get_property(msdo::download_property::id, downloadId2);
+    ASSERT_EQ(downloadId1, downloadId2);
+
+    download2->start_and_wait_until_completion();
```

Comment:
>start_and_wait_until_completion [](http://example.com/codeflow?start=15&length=31)

Probably should add a 1s wait after completion since job removal is async.